pkg/cloud/libvirt/client/ignition.go

```diff
@@ -43,16 +44,44 @@ func setIgnition(domainDef *libvirtxml.Domain, client *libvirtClient, ignition *
 		return err
 	}
 
-	domainDef.QEMUCommandline = &libvirtxml.DomainQEMUCommandline{
-		Args: []libvirtxml.DomainQEMUCommandlineArg{
-			{
-				// https://github.com/qemu/qemu/blob/master/docs/specs/fw_cfg.txt
-				Value: "-fw_cfg",
+	if strings.HasPrefix(arch, "s390") || strings.HasPrefix(arch, "ppc64") {
+		// System Z and PowerPC do not support the Firmware Configuration
+		// device. After a discussion about the best way to support a similar
+		// method for qemu in https://github.com/coreos/ignition/issues/928,
+		// decided on creating a virtio-blk device with a serial of ignition
+		// which contains the ignition config and have ignition support for
+		// reading from the device which landed in https://github.com/coreos/ignition/pull/936
```

Comment:
Isn't this code doing the same as calling `setCoreOSIgnition(ignitionVolumeName)`?

Reply:
oh yes - that would be a good optimization to make although this is a commit cherry-picked from master to 4.4. I can make that optimization to the master branch.

Reply:
opened https://github.com/openshift/cluster-api-provider-libvirt/pull/194 for the cleanup. meanwhile could this PR be approved please? thanks!